doc/vxlan/EVPN/EVPN_VXLAN_HLD.md

```diff
@@ -0,0 +1,1352 @@
+
+![Asymmetric IRB packet flow](images/pktflowAsymmIRB.PNG "Figure 2 : Asymm IRB Flow")
+__Figure 2: Asymmetric IRB packet flow__
+
+In the symmetric IRB model the VTEP is only configured with the subnets that are present on the directly attached hosts. Connectivity to non-local subnets on a remote VTEP is achieved through an intermediate IP-VRF.
+
+
+
+The ingress VTEP routes the traffic between the local subnet and the IP-VRF, which both VTEPs are a member of, the egress VTEP then routes the frame from the IP-VRF to the destination subnet.  The forwarding model results in both VTEPs performing a routing function.
+
+![Symmetric IRB packet flow](images/pktflowSymmIRB.PNG "Figure 3 : Symm IRB Flow")
+__Figure 3: Symmetric IRB packet flow__
+
+
+Both asymmetric and symmetric IRB models will be supported in SONiC.
+
+
+
+#### ARP and ND Suppression
+
+EVPN Type-2 routes make remote MAC-IP binding available on local device. This allows any ARP/ND request originated by local hosts for the remote IP to be serviced locally using the MAC-IP binding. This reduces ARP flooding in the network.
+
+
+
+
+# 4 Feature Design
+
+
+
+## 4.1 Design Overview
+
+### 4.1.1 Basic Approach
+
+For control plane, MP-BGP with EVPN extension support is leveraged from FRR with changes in Zebra and Fpmsyncd to interact with SONiC SwSS. A new process Fdbsyncd is added in SwSS to sync MAC route with kernel.  
+
+VXLAN orchagent and manager, FDB orchagent, VRF orchagent and manager, route orchagent   are enhanced to achieve this feature. 
+
+
+### 4.1.2 Container
+
+No new container is added. The changes are added to existing containers such as FRR, SwSS and Syncd. The details of the changes will be discussed in the Design section below.
+
+### 4.1.3 SAI Overview
+
+To support this feature, SAI will be extended as described in the SAI PRs below:
+
+- [Support for MAC Move](https://github.com/opencomputeproject/SAI/pull/1024)
+- [Support for L2VXLAN](https://github.com/opencomputeproject/SAI/pull/1025)
+- [Support for ARP/ND Suppression](https://github.com/opencomputeproject/SAI/pull/1056)
+
+## 4.2 DB Changes
+
+At a high level below are some of the interactions between relevant components and the DB involved for EVPN L2 VXLAN support in SONiC architecture.
+
+![EVPN L2 VXLAN Arch](images/arch.PNG "Figure : Arch")
+__Figure 4: EVPN L2 VXLAN DBs__
+
+### 4.2.1 CONFIG_DB changes
+
+**EVPN_NVO_TABLE**
+
+Producer:  config manager 
+
+Consumer: VxlanMgr
+
+Description: This is a singleton configuration which holds the tunnel name specifying the VTEP source IP used for BGP-EVPN discovered tunnels. 
+
+Schema:
+
+```
+;New table
+;holds the tunnel name specifying the VTEP source IP used for BGP-EVPN discovered tunnels
+
+key = VXLAN_EVPN_NVO|nvo_name
+                          ; nvo or VTEP name as a string
+; field = value
+source_vtep = tunnel_name ; refers to the tunnel object created with SIP only.
+```
+
```

Comment:
Can we have an example of config_db, similar to https://github.com/prsunny/SONiC/blob/f85f2b178743343b2c26d0a3e21392bdaae9745f/doc/vxlan/Vxlan_hld.md#4-example-configuration

Reply:
Will give an example.